dbt/models/default/default.vw_pin_status.sql

```diff
@@ -87,6 +87,29 @@ WITH ahsap AS (
     WHERE par.cur = 'Y'
         AND par.deactivat IS NULL
     GROUP BY par.parid, par.taxyr
+),
+
+assessable_permits_past_3_years AS (
+    SELECT
+        pardat.parid AS pin,
+        pardat.taxyr AS year,
+        SUM(
+            CASE WHEN permit.year IS NULL THEN 0 ELSE 1 END
+        ) AS num_assessable_permits
+    FROM {{ source('iasworld', 'pardat') }} AS pardat
+    LEFT JOIN (
+        SELECT
+            pin,
+            SUBSTR(date_issued, 1, 4) AS year
+        FROM {{ ref('default.vw_pin_permit') }}
+        WHERE assessable = 'A'
+            AND date_issued IS NOT NULL
+        GROUP BY pin, SUBSTR(date_issued, 1, 4)
+    ) AS permit
+        ON pardat.parid = permit.pin
+        AND CAST(permit.year AS INT)
+        BETWEEN CAST(pardat.taxyr AS INT) - 2 AND CAST(pardat.taxyr AS INT)
+    GROUP BY pardat.parid, pardat.taxyr
```

Comment:
This query looks a bit gnarly, but I tested it against the current prod view and confirmed that it doesn't change the runtime or amount of queried data by very much.